mediapipe/modules/face_geometry/libs/geometry_pipeline_with_attention.cc

```diff
@@ -0,0 +1,458 @@
+    right = 0.5f * width_at_near;
+    bottom = -0.5f * height_at_near;
+    top = 0.5f * height_at_near;
+    near = perspective_camera.near();
+    far = perspective_camera.far();
+  }
+
+  float left;
+  float right;
+  float bottom;
+  float top;
+  float near;
+  float far;
+};
+
+class ScreenToMetricSpaceConverter {
+ public:
+  ScreenToMetricSpaceConverter(
+      OriginPointLocation origin_point_location,      //
+      InputSource input_source,                       //
+      Eigen::Matrix3Xf&& canonical_metric_landmarks,  //
+      Eigen::VectorXf&& landmark_weights,             //
+      std::unique_ptr<ProcrustesSolver> procrustes_solver)
+      : origin_point_location_(origin_point_location),
+        input_source_(input_source),
+        canonical_metric_landmarks_(std::move(canonical_metric_landmarks)),
+        landmark_weights_(std::move(landmark_weights)),
+        procrustes_solver_(std::move(procrustes_solver)) {}
+
+  // Converts `screen_landmark_list` into `metric_landmark_list` and estimates
+  // the `pose_transform_mat`.
+  //
+  // Here's the algorithm summary:
+  //
+  // (1) Project X- and Y- screen landmark coordinates at the Z near plane.
+  //
+  // (2) Estimate a canonical-to-runtime landmark set scale by running the
+  //     Procrustes solver using the screen runtime landmarks.
+  //
+  //     On this iteration, screen landmarks are used instead of unprojected
+  //     metric landmarks as it is not safe to unproject due to the relative
+  //     nature of the input screen landmark Z coordinate.
+  //
+  // (3) Use the canonical-to-runtime scale from (2) to unproject the screen
+  //     landmarks. The result is referenced as "intermediate landmarks" because
+  //     they are the first estimation of the resuling metric landmarks, but are
+  //     not quite there yet.
+  //
+  // (4) Estimate a canonical-to-runtime landmark set scale by running the
+  //     Procrustes solver using the intermediate runtime landmarks.
+  //
+  // (5) Use the product of the scale factors from (2) and (4) to unproject
+  //     the screen landmarks the second time. This is the second and the final
+  //     estimation of the metric landmarks.
+  //
+  // (6) Multiply each of the metric landmarks by the inverse pose
+  //     transformation matrix to align the runtime metric face landmarks with
+  //     the canonical metric face landmarks.
+  //
+  // Note: the input screen landmarks are in the left-handed coordinate system,
+  //       however any metric landmarks - including the canonical metric
+  //       landmarks, the final runtime metric landmarks and any intermediate
+  //       runtime metric landmarks - are in the right-handed coordinate system.
+  //
+  //       To keep the logic correct, the landmark set handedness is changed any
+  //       time the screen-to-metric semantic barrier is passed.
+  absl::Status ConvertWithAttention(const NormalizedLandmarkList& screen_landmark_list,  //
+      bool with_attention,
```

Comment:
As I understand major change of this file is due to different landmarks number, maybe we can create a special constant for the landmarks number (in some separate file)?

Reply:
File which would be used for one constant?

Reply:
No, I try to find a way to simplify such changes connected with attention (different landmarks size), but I guess there is no good way to fix it.

Reply:
So, I resolve this.